src/AstValidationSegmenter.ts

```diff
@@ -77,8 +77,11 @@ export class AstValidationSegmenter {
                 this.checkExpressionForUnresolved(segment, expression.expression.right as VariableExpression, assignedSymbolsNames);
         }
         if (isTypeExpression(expression)) {
-            const typeIntypeExpression = expression.getType({ flags: SymbolTypeFlag.typetime });
-            if (typeIntypeExpression.isResolvable()) {
+            let typeInTypeExpression = expression.getType({ flags: SymbolTypeFlag.typetime });
+            if (isArrayType(typeInTypeExpression)) {
+                typeInTypeExpression = typeInTypeExpression.defaultType;
```

Comment:
make sure we're actually checking the default type of the the array for being unresolved.